Zend/zend_language_parser.y

```diff
@@ -595,6 +600,32 @@ case_separator:
 ;
 
 
+match_arm_list:
+		%empty { $$ = zend_ast_create_list(0, ZEND_AST_MATCH_ARM_LIST); }
```

Comment:
I guess this is the most consistent with switch and a reasonable base case (e.g. for automatically generated code). A php linter or analyzer could easily be set up to catch code not providing any cases for switches or cases, which should be extremely infrequent in practice

```
php > switch(2){}  // no compile warnings or notices
php > match(2){}
Warning: Uncaught UnhandledMatchError in php shell code:1
```

Reply:
Alternatively we could throw a compiler error when the match is empty. That would also make sense. I just didn't want it to be a parser error to avoid confusion.

Reply:
I think this is fine for now. 🙂